src/node_api_types.h

```diff
@@ -82,9 +65,33 @@ typedef enum {
   napi_array_expected,
   napi_generic_failure,
   napi_pending_exception,
+  napi_cancelled,
   napi_status_last
 } napi_status;
 
+typedef void (*napi_callback)(napi_env env, 
+                              napi_callback_info info);
+typedef void (*napi_finalize)(napi_env env,
+                              void* finalize_data, 
+                              void* finalize_hint);
+typedef void (*napi_async_execute_callback)(napi_env env, 
+                                            void* data);
+typedef void (*napi_async_complete_callback)(napi_env env,
+                                             napi_status status,
+                                             void* data);
+
+typedef struct {
+  const char* utf8name;
+
+  napi_callback method;
+  napi_callback getter;
+  napi_callback setter;
+  napi_value value;
+
+  napi_property_attributes attributes;
+  void* data;
+} napi_property_descriptor;
```

Comment:
I put the `napi_callback` typedef near `napi_async_execute_callback` and other callback typedefs. But `napi_async_complete_callback` takes a `napi_status` so all the callbacks should be located down below the enum (at least). However, `napi_property_descriptor` has `napi_callback` members so it also had to move down in the file.